### DISCUSSION TITLE
Optimizations & Kernels

### DISCUSSION THREAD

**Discussion by MortThePersian** [XTLS/Xray-core#1984] [General]
Dear My Sisters & Brothers;

You do probably heard or even used some tools to optimise the VPS.

Some of them are more clear than others like "bbr".

The question I have encountered is about the kernels that I saw here & there, 'Zen' & 'Xanmod' kernels in the tools related to v2ray

Despite the various attempts, No clear answer I got in this regard...

What are the benefits these kernels have for Us ?!

I also exceedingly curious to know the different impacts of, Generic, Low Latency & Real-Time Kernels on Our use case scenario.

Best Regards

**Comment by ashad765** [comment]
I tried many different kernel version and different tcp optimization like bbrplus and some Chinese names that I can’t even pronounce correctly and in most case stability become a serious problem. If you have time you can try different thing and find out best solution for your need, but from my experience only two things really matter

1. Set tcp congestion control to bbr and Tune the kernel parameters like increasing the maximum number of open file descriptors and Enlarging the queues and buffers
2. Find a good line

And the second is much more important than the first. For example hetzner was a great choice for Iran but now most of their new IP got blocked quickly

**Reply by MortThePersian** [reply]
Thank you for sharing your experiences 

As I mentioned earlier recently in regards to current Iran status, Situation complexity rised to the level even inside of a city users report different kinds of restrictions through the various providers !

https://github.com/XTLS/Xray-core/discussions/1961

But since you mentioned about your tests & experiences, Have you ever tested Real-Time Kernels?! 
Have a look at this for example;

https://ubuntu.com/blog/real-time-ubuntu-is-now-generally-available

**Reply by ashad765** [reply]
I didn't test it yet but as I said you can test it yourself and find the result, but It probably won't make a game changer different.

As you mention the current situation in Iran become very complex. If you want to run your own personal server you'll be fine in most cases, but finding good solution for selling proxy are very difficult. Overal you should choose between CDN or direct connection.

1. If you choose CDN cloudflare become the obvious choice and also the most problematic one. most of CF IP are blocked and it becomes nearly useless right now, so you should try other CDNs.I had friends that running server behind belugaCDN and it worked perfectly fine. The problem will be price as it cost 150$ for 10TB monthly data plan.

2. If you choose direct connection then reality will become best solution, but your server will be blocked if you transfer for example 10TB per month for sure, so choosing the right vps provider and server name become critical. My friends running server on Akamai and using Apple cdn for server name and it worked fine for more than 2 month without any problem at 5 TB monthly plan. Personally I can't verify it but choosing Google vps and using google subdomain like dl.google.com also should work fine for selling purpose.

**Reply by kalilovers** [reply]
how we can optimize bbr config to best mode ?buffer and ...

**Comment by us254** [comment]
- Use the following command to edit the `sysctl.conf` file using nano.

    
    sudo nano /etc/sysctl.conf

3. **Add Settings:**
   - Scroll to the bottom of the file and add your networking parameter settings. For instance:

    ```conf
    # BBR TCP Congestion Control Optimization
    net.core.rmem_max=2500000
    net.core.wmem_max=2500000
    net.ipv4.tcp_rmem=4096 87380 2500000
    net.ipv4.tcp_wmem=4096 65536 2500000
    net.ipv4.tcp_congestion_control=bbr
    ```

4. **Save Changes:**
   - Press `CTRL + O` to save changes to the file, then press `Enter`.

5. **Exit Nano:**
   - Press `CTRL + X` to exit nano.

6. **Apply Changes Without Rebooting:**
   - To apply the changes without rebooting, execute the following command:

    ```bash
    sudo sysctl -p
    ```

   This command will reload the settings from your `sysctl.conf` file.

**Reply by kalilovers** [reply]
thanks alot . i mean this was best parameter setting for default kernel bbr config ?

**Comment by hawshemi** [comment]
**https://github.com/hawshemi/Linux-Optimizer**
